### DISCUSSION TITLE
Popover usage - any way to access the close event?

### DISCUSSION THREAD

**Discussion by gtiminskis-sc** [ariakit/ariakit#1313] [Questions] [answered]
Hello!
First of all I wanted to thank you on the hard work creating this library, it's been a real joy to use.

I have a pretty basic question - is there any way to access an event for visibility change of the `Popover` component?
I want to manually control the open/closed state of the Popover dropdown (using the `visible` prop) but I can't figure out a way to know when the user has clicked off the dropdown (which triggers the change internally).

Thank you!

**Comment by diegohaz** [comment, marked as answer]
I'm not sure I fully understand your use case (a code sample or CodeSandbox would be nice). Anyway, I'll assume you're abstracting the dialog component and want to control the dialog state from outside.

On v1 it's a little bit verbose (https://codesandbox.io/s/reakit-dialog-ontoggle-zw0683).

That's one of the things v2 improves. On v2, you can do this:
```jsx
function MyDialog({ isOpen, onToggle }) {
  const dialog = useDialogState({ visible: isOpen, setVisible: onToggle });
  ...
}
```

If you want specific events (interact outside, escape):
```jsx
function MyDialog({ onClickOutside, onEscape }) {
  ...
  <Dialog
    hideOnInteractOutside={(event) => {
      onClickOutside(event);
      return true;
    }}
    hideOnEscape={(event) => {
      onEscape(event);
      return true;
    }}
  />
}
```

**Reply by gtiminskis-sc** [reply]
Sorry forgot to specify :) Currently using v1

**Reply by diegohaz** [reply]
Updated the comment.
